homeassistant/components/weatherkit/weather.py

```diff
@@ -0,0 +1,255 @@
+    "IsolatedThunderstorms": "lightning",
+    "Rain": "rainy",
+    "SunShowers": "rainy",
+    "ScatteredThunderstorms": "lightning",
+    "StrongStorms": "lightning",
+    "Thunderstorms": "lightning",
+    "Frigid": "snowy",
+    "Hail": "hail",
+    "Hot": "sunny",
+    "Flurries": "snowy",
+    "Sleet": "snowy",
+    "Snow": "snowy",
+    "SunFlurries": "snowy",
+    "WintryMix": "snowy",
+    "Blizzard": "snowy",
+    "BlowingSnow": "snowy",
+    "FreezingDrizzle": "snowy-rainy",
+    "FreezingRain": "snowy-rainy",
+    "HeavySnow": "snowy",
+    "Hurricane": "exceptional",
+    "TropicalStorm": "exceptional",
+}
+
+
+def _map_daily_forecast(forecast) -> Forecast:
+    return {
+        "datetime": forecast.get("forecastStart"),
+        "condition": condition_code_to_hass[forecast.get("conditionCode")],
+        "native_temperature": forecast.get("temperatureMax"),
+        "native_templow": forecast.get("temperatureMin"),
+        "native_precipitation": forecast.get("precipitationAmount"),
+        "precipitation_probability": forecast.get("precipitationChance") * 100,
+        "uv_index": forecast.get("maxUvIndex"),
+    }
+
+
+def _map_hourly_forecast(forecast) -> Forecast:
+    return {
+        "datetime": forecast.get("forecastStart"),
+        "condition": condition_code_to_hass[forecast.get("conditionCode")],
+        "native_temperature": forecast.get("temperature"),
+        "native_apparent_temperature": forecast.get("temperatureApparent"),
+        "native_dew_point": forecast.get("temperatureDewPoint"),
+        "native_pressure": forecast.get("pressure"),
+        "native_wind_gust_speed": forecast.get("windGust"),
+        "native_wind_speed": forecast.get("windSpeed"),
+        "wind_bearing": forecast.get("windDirection"),
+        "humidity": forecast.get("humidity") * 100,
+        "native_precipitation": forecast.get("precipitationAmount"),
+        "precipitation_probability": forecast.get("precipitationChance") * 100,
+        "cloud_coverage": forecast.get("cloudCover") * 100,
+        "uv_index": forecast.get("uvIndex"),
+    }
+
+
+class WeatherKitWeather(
+    SingleCoordinatorWeatherEntity[WeatherKitDataUpdateCoordinator]
+):
+    """Weather entity for Apple WeatherKit integration."""
+
+    _attr_attribution = ATTRIBUTION
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    _attr_native_temperature_unit = UnitOfTemperature.CELSIUS
+    _attr_native_pressure_unit = UnitOfPressure.MBAR
+    _attr_native_visibility_unit = UnitOfLength.KILOMETERS
+    _attr_native_wind_speed_unit = UnitOfSpeed.KILOMETERS_PER_HOUR
+
+    def __init__(
+        self,
+        coordinator: WeatherKitDataUpdateCoordinator,
+    ) -> None:
+        """Initialise the platform with a data instance and site."""
+        super().__init__(coordinator)
+        self._config = coordinator.config_entry.data
+        self._attr_unique_id = (
+            f"{self._config[CONF_LATITUDE]}-{self._config[CONF_LONGITUDE]}"
+        )
```

Comment:
```suggestion
        config_data = coordinator.config_entry.data
        self._attr_unique_id = (
            f"{config_data[CONF_LATITUDE]}-{config_data[CONF_LONGITUDE]}"
        )
```
Looks like you're never using `self._config` outside of the constructor.

Reply:
Ah, yeah. I think I used it elsewhere in a previous revision but forgot to remove it